ts/utils/serve_decorator.py

```diff
@@ -0,0 +1,215 @@
+
+    inference() can be as complex as you like and call other functions 
+
+    If you already have a handler you can manually call create_torchserve_config(), archive_model() 
+    and start_torchserve()
+
+    What this will do is
+    1. Override the base handler with a new handler with the inference function overloaded
+    2. Create a new torchserve config 
+    3. Archive a model assuming weights are in disk
+    4. Start torchserve
+
+    # TODO: Put everything in a temporary directory so user files don't get overwritten
+    """
+    @wraps(func)
+    def serve_model(*args, **kwargs):
+        """
+        Can override inference and management port or model name here
+        @serve(inference_http_port=8080, management_port=8081, serialized_file="my_model.pt", batch_size=32)
+        Full available configurations can be found in ts_config`
+
+        If you need to serve a full handler you can instead call create_handler, create_torchserve_config and torchserve_start manually
+        """
+        
+        handler = create_handler(func)
+
+        config = create_torchserve_config(inference_http_port=7070, management_http_port=7071, batch_size=32)
+        
+        model = archive_model(serialized_file="model.pt")
+
+        torchserve_start(handler= handler, models=model, config=config)
+
+        logger.warning("Don't forget to call torchserve --stop when you're done")
+        
+        return 0
+
+
+def create_handler(inference_func : Union[Callable, BaseHandler], handler_file : str = "handler.py"):
+    """
+    Create a new handler that inherits from the base handler
+    Update the inference() function with the function captured by the decorator
+    Write the handler to disk
+
+    TODO: Improve story for adding custom dependencies, all the hidden imports or library code
+    TODO: Improve workflow story, coming in future PR
+
+    """
+    
+    # If user provides a function then override inference() function in base handler
+    if type(inference_func) is Callable:
+        handler_class = BaseHandler()
+        handler_class.inference = inference_func
+    
+    # If user provides a class then create a new custom handler
+    else:
+        handler_class = inference_func
+
+    # Get source code for new class and write it to disk
+    source = inspect.getsource(handler_class)
+    with open(handler_file, "w") as f:
+        f.write(source)
+    
+    return handler_file
+
+def writetofile(filename : str = "model.py"):
+    """
+    @writetofile decorator to add on top of a torch.nn.Module to write it to disk as model.py
+    Use this instead on top of another handler
+    """
+    @wraps(cls)
+    def archive_model(*args, **kwargs):
+        class_source = inspect.getsource(cls)
+        with open(filename, "w") as f:
+            f.write(class_source)
+
+
+def torchserve_start(handler : str, model_store : str ="model_store", ts_config : str ="config.properties", model_mar : str ="model.mar"):
```

Comment:
I wonder if using existing utils in [ts_script](https://github.com/pytorch/serve/blob/master/ts_scripts/tsutils.py#L27) would be helpful/ avoid redundancy? or alternatively expand it to cover workflows( in case we re deprecating yet).

Reply:
Sure lemme me how to refactor

For workflows specifically this is not coming soon so will not address in this PR

Reply:
`ts_script` is not part of the torchserve library, it only includes what's in the `ts` code. I can refactor this in a future PR